### PR TITLE
Update local development setup instructions in CONTRIBUTING.md and template

### LINKED ISSUE #203: Incorrect Setup Instructions in CONTRIBUTING for Local Development
### Overview
The current setup instructions provided in the CONTRIBUTING.md for local development are outdated and incorrect. The instructions suggest using `python setup.py develop` for installation, but the `setup.py` file does not exist in the repository. This issue could also help in fixing the CONTRIBUTING.md in the template too

### Proposed Solution:
Update the CONTRIBUTING file with accurate and up-to-date instructions for local development. This may involve recommending alternative methods for installation, such as using Poetry `poetry install` or pip `pip install -e .`.

### Documentation Page Link

https://osl-incubator.github.io/scicookie/contributing.html#get-started

### Code of Conduct

- [X] I agree to follow the Code of Conduct

### PR DESCRIPTION
**Pull request by Naman-Priyadarshi**
## Pull Request description
This pull request updates the CONTRIBUTING file for scicookie and the template to provide accurate and up-to-date instructions for setting up the project locally. The previous instructions were outdated and misleading, as they referenced a setup.py file that does not exist in the repository.

Changes Made:

Updated the CONTRIBUTING.md with clear and concise instructions for local development setup.
Removed references to the nonexistent setup.py file.
Provided alternative installation methods such as Poetry (poetry install) and pip (pip install -e .).
Replaced `$` with Code Blocks
Fixed Issue in template where instructions said tox and flake8 but referenced `make lint` and `make tests`

 Fixes #203 

## How to test these changes

* ```...```

## Pull Request checklists

This PR is a:
- [ ] bug-fix
- [ ] new feature
- [x] maintenance

About this PR:
- [ ] it includes tests.
- [ ] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [x] pre-commit hooks were executed locally.
- [ ] this PR requires a project documentation update.

Author's checklist:
- [ ] I have reviewed the changes and it contains no misspelling.
- [ ] The code is well commented, especially in the parts that contain more complexity.
- [x] New and old tests passed locally.

## Reviewer's checklist

Copy and paste this template for your review's note:

```
## Reviewer's Checklist

- [ ] I managed to reproduce the problem locally from the `main` branch
- [ ] I managed to test the new changes locally
- [ ] I confirm that the issues mentioned were fixed/resolved
```

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -54,36 +54,48 @@ Ready to contribute? Here’s how to set up `scicookie` for local development.
...
-    $ python setup.py develop
-
-4.  Create a branch for local development::
-
-    $ git checkout -b name-of-your-bugfix-or-feature
-
+    local development and this will automatically install all required and `dev` dependencies: 
+    ```
+    mkvirtualenv scicookie
+    cd scicookie
```
Comment (Saransh-cpp) [inline CONTRIBUTING.md:66]: Let's use virtualenv directly
```suggestion
    cd scicookie
    python -m venv env
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -54,36 +54,48 @@ Ready to contribute? Here’s how to set up `scicookie` for local development.
...
 
 5.  When you’re done making changes, check that your changes pass flake8
-    and the tests, including testing other Python versions with tox::
-
-    $ flake8 scicookie tests $ python setup.py
-    test or pytest $ tox
-
+    and the tests, including testing other Python versions with tox:
+    ```
+    flake8 scicookie tests
```
Comment (Saransh-cpp) [inline CONTRIBUTING.md:87]:
```suggestion
    pre-commit install   # will install pre-commit hooks to run before every commit 
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -102,7 +114,7 @@ Before you submit a pull request, check that it meets these guidelines:
 To run a subset of tests::
 
 ```
-$ pytest
+pytest
```
Comment (Saransh-cpp) [inline CONTRIBUTING.md:117]:
```suggestion
pytest tests.<the test to run>
```

File: src/scicookie/{{cookiecutter.project_slug}}/docs/contributing.md [hunk 4]
```diff
@@ -119,36 +119,50 @@ Ready to contribute? Here’s how to set up `{{ cookiecutter.project_slug}}` for
 
 1.  Fork the `{{ cookiecutter.project_slug }}` repo on GitHub.
```
Comment (Saransh-cpp) [inline src/scicookie/{{cookiecutter.project_slug}}/docs/contributing.md:120]: Ditto for this file

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
Thanks for taking this up, @Naman-Priyadarshi! A few comments below -

**Comment by Naman-Priyadarshi** [comment]
I have resolved the changes. Also, I replaced the flake8 and tox step with adding a pre-commit section

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]

**Pull request merged**